Removes vendor and updates google.golang.org/api to v0.83.0

austingebauer opened the pull request:
This PR removes the vendor directory and updates the `google.golang.org/api` dependency to `v0.83.0`.

This will be tested in the following locations before merging:
- [x] https://github.com/hashicorp/vault
- [x] https://github.com/hashicorp/vault-plugin-auth-gcp
- [x] https://github.com/hashicorp/vault-plugin-secrets-gcp

austingebauer committed:
Removes vendor and updates google.golang.org/api to v0.83.0

calvn commented on README.md:
```diff
@@ -1,17 +1,16 @@
 
 This library was created to be shared by Vault-GCP integrations including the:
-- [GCP Auth Plugin](https://github.com/emilymye/vault-plugin-auth-gcp)
+- [GCP Auth Method](https://github.com/emilymye/vault-plugin-auth-gcp)
+- [GCP Secrets Engine](https://github.com/emilymye/vault-plugin-secrets-gcp)
```
Do we want to link the repos from the `hashicorp/` org instead?

austingebauer replied:
Whoops! Done in https://github.com/hashicorp/go-gcp-common/pull/8/commits/dbeaedcf7363b486d2936133dc381d081f8fbea8 👍

austingebauer committed:
link to repos from hashicorp org

austingebauer commented:
Can confirm that the acceptance tests passed in both GCP auth and secrets plugins. Testing in Vault next.

calvn commented:
How disruptive would the changeset be (compared to now) if you were to also bump the go version in go.mod?

austingebauer commented:
@calvn - I don't think it would be too disruptive at this point, but I'm also not fully aware of the implications. Any specific min version you were thinking to bump it to?

Looks like it's set to 1.12 in [vault-plugin-secrets-gcp/blob/main/go.mod#L3](https://github.com/hashicorp/vault-plugin-secrets-gcp/blob/main/go.mod#L3), [vault-plugin-auth-gcp/blob/main/go.mod#L3](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/main/go.mod#L3) as well.

calvn commented:
I don't have a specific version in mind but 1.17 or 1.18 would be good, since that's when [they changed](https://go.dev/doc/go1.17#:~:text=Pruned%20module%20graphs%20in%20go%201.17%20modules) the formatting on how they group and include direct and indirect dependencies. Not a big deal if we leave if out of this PR, was just mostly curious. 

We may also want to add `hashicorp/vault` to that list of repos that use this library.

austingebauer committed:
Update go.mod go version to 1.17; Add Vault link to readme

austingebauer commented:
@calvn - I went ahead and updated it to Go 1.17. I also added the Vault link to the readme. I was able to successfully test this in Vault for GCS storage, Agent GCP auto-auth, GCP secrets, and GCP auth.

calvn commented:
Thanks, appreciate the testing and validation done against those repos!
